src/UsgsAstroLsSensorModel.cpp

```diff
@@ -2727,60 +2722,51 @@ std::string UsgsAstroLsSensorModel::constructStateFromIsd(const std::string imag
    state["m_mountingMatrix"][7] = sin_a * cos_b;
    state["m_mountingMatrix"][8] = cos_a * cos_b;
 
-   state["m_dtEphem"] = isd.at("DT_EPHEM");
-   state["m_t0Ephem"] = isd.at("T0_EPHEM");
-   state["m_dtQuat"] =  isd.at("DT_QUAT");
-   state["m_t0Quat"] =  isd.at("T0_QUAT");
-
-   state["m_numEphem"] = isd.at("NUMBER_OF_EPHEM");
-   state["m_numQuaternions"] = isd.at("NUMBER_OF_QUATERNIONS");
-
-   int numEphem = isd.at("NUMBER_OF_EPHEM");
-   for (int i=0;i < numEphem * 3; i++){
-       state["m_ephemPts"].push_back(isd.at("EPHEM_PTS").at(i));
-       state["m_ephemRates"].push_back(isd.at("EPHEM_RATES").at(i));
-   }
-
-   int numQuat = isd.at("NUMBER_OF_QUATERNIONS");
-   for (int i=0; i < numQuat * 4; i++){
-       state["m_quaternions"].push_back(isd.at("QUATERNIONS").at(i));
+   state["m_dtEphem"] = isd.at("dt_ephemeris");
+   state["m_t0Ephem"] = isd.at("t0_ephemeris");
+   state["m_dtQuat"] =  isd.at("dt_ephemeris");
+   state["m_t0Quat"] =  isd.at("t0_ephemeris");
+
+   state["m_numEphem"] = isd.at("number_of_ephemerides");
+   state["m_numQuaternions"] = isd.at("number_of_ephemerides");
+
+   int numEphem = isd.at("number_of_ephemerides");
+   for (int i=0;i < numEphem; i++){
+       state["m_ephemPts"].push_back(isd.at("sensor_location").at("x").at(i));
+       state["m_ephemPts"].push_back(isd.at("sensor_location").at("y").at(i));
+       state["m_ephemPts"].push_back(isd.at("sensor_location").at("z").at(i));
+       state["m_ephemRates"].push_back(isd.at("sensor_velocity").at("x").at(i));
+       state["m_ephemRates"].push_back(isd.at("sensor_velocity").at("y").at(i));
+       state["m_ephemRates"].push_back(isd.at("sensor_velocity").at("z").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("w").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("v1").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("v2").at(i));
+       state["m_quaternions"].push_back(isd.at("sensor_orientation").at("v3").at(i));

```

Comment:
A bigger question - when might one not have number_of_ephemerides and number_of_quaternions not in sync? If the answer is never, we can drop number_of_quaternions from the spec.